src/code/z_game_over.c

```diff
@@ -0,0 +1,150 @@
+#include "global.h"
+
+void GameOver_Init(GlobalContext* globalCtx) {
+    globalCtx->gameOverCtx.state = GAMEOVER_INACTIVE;
+}
+
+void GameOver_FadeInLights(GlobalContext* globalCtx) {
+    GameOverContext* gameOverCtx = &globalCtx->gameOverCtx;
+
+    if ((gameOverCtx->state >= GAMEOVER_DEATH_WAIT_GROUND && gameOverCtx->state < GAMEOVER_REVIVE_START) ||
+        (gameOverCtx->state >= GAMEOVER_REVIVE_RUMBLE && gameOverCtx->state < GAMEOVER_REVIVE_FADE_OUT)) {
+        Kankyo_FadeInGameOverLights(globalCtx);
+    }
+}
+
+// This variable has the same problem as z_message's data going to rodata.
+// A fix for this is planned, and will be taken care of with z_message.
+// For now this variable is externed from z_message's rodata.s file.
+extern s16 D_80153D80; // todo: rename to `timer` and make this in function static (after rodata issue is resolved)
+
+void GameOver_Update(GlobalContext* globalCtx) {
+    GameOverContext* gameOverCtx = &globalCtx->gameOverCtx;
+    s16 i;
+    s16 j;
+    s32 v90;
+    s32 v91;
+    s32 v92;
+
+    switch (gameOverCtx->state) {
+        case GAMEOVER_DEATH_START:
+            func_80106CCC(globalCtx);
+
+            gSaveContext.timer1State = 0;
+            gSaveContext.timer2State = 0;
+            gSaveContext.eventInf[1] &= ~1;
+
+            for (i = 0; i < ARRAY_COUNT(gSpoilingItems); i++) { // search for spoiling items
+                if (INV_CONTENT(ITEM_POCKET_EGG) == gSpoilingItems[i]) {
+                    INV_CONTENT(gSpoilingItemReverts[i]) =
+                        gSpoilingItemReverts[i]; // if a spoiling item is found, revert it
+                    for (j = 1; j < ARRAY_COUNT(gSaveContext.equips.buttonItems);
+                         j++) { // search c buttons for the found spoiling item
+                        if (gSaveContext.equips.buttonItems[j] == gSpoilingItems[i]) {
+                            gSaveContext.equips.buttonItems[j] =
+                                gSpoilingItemReverts[i]; // if found on a c button, revert it
```

Comment:
Formatter is breaking up these various lines because of the comments, so I suggest putting them in the lines above